force-app/main/default/API/classes/RestServices/CRM_HenvRedactionRestService.cls

```diff
@@ -0,0 +1,98 @@
+    global static void markForRedaction() {
+        String navIdent = getActingNavIdent();
+        correlationId = getXCorrelation();
+        logger.apiTransaction(correlationId, RestContext.request, CRM_ApplicationDomain.Domain.NKS);
+        try {
+            RestRequest request = RestContext.request;
+            SladdingRequest slRequest = (SladdingRequest) JSON.deserialize(
+                request.requestBody.toString(),
+                SladdingRequest.class
+            );
+
+            Id objectId = getObjectId(slRequest.kjedeId);
+            String objectName = objectId != null ? objectId.getSobjectType().getDescribe().getName() : null;
+
+            if (objectName == 'Thread__c' || objectName == 'Conversation_Note__c') {
+                SObject objToUpdate = objectId.getSobjectType().newSObject(objectId);
+                objToUpdate.put('STO_Sensitive_Information__c', true);
+                objToUpdate.put('Cause__c', slRequest.aarsak);
+                objToUpdate.put('CRM_Last_Edited_By__c', navIdent);
+
+                update objToUpdate;
+                setResponse(200);
+            } else {
+                errorResponse(400, 'Invalid reference value for kjedeId: ' + slRequest.kjedeId);
+                logger.publish();
+                return;
+            }
+        } catch (DmlException dmlEx) {
+            errorResponse(500, dmlEx.getMessage());
+            logger.error('DML EXCEPTION: ' + dmlEx.getMessage(), null, CRM_ApplicationDomain.Domain.NKS, correlationId);
+        } catch (Exception ex) {
+            errorResponse(500, 'Failed to update: ' + ex.getMessage());
+            logger.error(
+                'Failed to update: ' +
+                ex.getMessage() +
+                '\n' +
+                ex.getStackTraceString(),
+                null,
+                CRM_ApplicationDomain.Domain.NKS,
+                correlationId
+            );
+        }
+    }
+
+    /**
+     * Returns the available redaction causes for the given kjedeId
+     * @author Stian Ruud Schikora | 10-04-2022
+     **/
+    @HttpGet
+    global static void getRedactionCauses() {
+        correlationId = getXCorrelation();
+        logger.apiTransaction(correlationId, RestContext.request, CRM_ApplicationDomain.Domain.NKS);
+        String kjedeId = getLastPathParameter();
+
+        Id objectId = getObjectId(kjedeId);
+        String objectName = objectId != null ? objectId.getSobjectType().getDescribe().getName() : null;
+
+        switch on objectName {
+            when 'Thread__c' {
+                List<Schema.PicklistEntry> causes = Thread__c.Cause__c.getDescribe().getPicklistValues();
+                setResponse(200, JSON.serializePretty(simplifyCauses(causes)));
+            }
+            when 'Conversation_Note__c' {
+                List<Schema.PicklistEntry> causes = Conversation_Note__c.Cause__c.getDescribe().getPicklistValues();
+                setResponse(200, JSON.serializePretty(simplifyCauses(causes)));
```

Comment:
Jeg ser ikke hvorfor denne trenger en kjedeId, den sjekker jo bare alle picklistverdiene for et objekt. Hadde det ikke vært bedre å bare bruke statiske strings som "tråd" og "samtalereferat"

Reply:
Hmmm, er det bare for å få et sentralisert endepunkt for sladding?

Reply:
Vet ikke om noen måte å vise en flow lokalt så hvis du finner noe vil jeg gjerne vite om det!